core/src/main/java/org/elasticsearch/index/analysis/TokenFilterFactory.java

```diff
@@ -20,10 +20,20 @@
 package org.elasticsearch.index.analysis;
 
 import org.apache.lucene.analysis.TokenStream;
+import org.apache.lucene.analysis.tokenattributes.OffsetAttribute;
+import org.elasticsearch.search.fetch.subphase.highlight.FastVectorHighlighter;
 
 public interface TokenFilterFactory {
-
     String name();
 
     TokenStream create(TokenStream tokenStream);
+
+    /**
+     * Does this analyzer mess up the {@link OffsetAttribute}s in such as way as to break the
+     * {@link FastVectorHighlighter}? If this is {@code true} then the
+     * {@linkplain FastVectorHighlighter} will attempt to work around the broken offsets.
+     */
+    default boolean breaksFastVectorHighlighter() {
```

Comment:
I'm not sure. I kind of like that the hack is at least more visible this way. For now I think we should keep it. Maybe we can pitch it if we ever go to 100% unified highlighter....